packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -101,35 +102,66 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
-        // The network may be only temporarily unavailable, or the file will be
-        // added on the server later. Avoid having future calls to resolve
-        // fail to check the network again.
-        await response.drain<List<int>>(<int>[]);
-        throw image_provider.NetworkImageLoadException(statusCode: response.statusCode, uri: resolved);
-      }
-
-      final Uint8List bytes = await consolidateHttpClientResponseBytes(
-        response,
-        onBytesReceived: (int cumulative, int? total) {
-          chunkEvents.add(ImageChunkEvent(
-            cumulativeBytesLoaded: cumulative,
-            expectedTotalBytes: total,
-          ));
-        },
-      );
-      if (bytes.lengthInBytes == 0) {
-        throw Exception('NetworkImage is an empty file: $resolved');
-      }
-
+      final RawReceivePort port = RawReceivePort();
+      final Completer<Uint8List> pendingBytes = Completer<Uint8List>();
+      port.handler = (Object? response) {
+        if (response is ImageChunkEvent) {
+          chunkEvents.add(response);
+        } else if (response is List<Object>) {
+          assert(() {
+            debugLastHttpClientUsed = response[1] as HttpClient;
+            return true;
+          }());
+          final Object data = response[0];
+          if (data is Uint8List) {
+            pendingBytes.complete(data);
+          } else {
+            pendingBytes.completeError(data);
+          }
+        } else {
+          assert(false);
+        }
+      };
+
+      // This code does not use compute because it needs to forward chunk events
+      // back to the main isolate.
+      Isolate.spawn((_LoadRequest data) async {
+        try {
+          final Uri resolved = Uri.base.resolve(data.url);
+          final HttpClientRequest request = await data.client.getUrl(resolved);
+          data.headers?.forEach(request.headers.add);
+          final HttpClientResponse response = await request.close();
+          if (response.statusCode != HttpStatus.ok) {
+            // The network may be only temporarily unavailable, or the file will be
+            // added on the server later. Avoid having future calls to resolve
+            // fail to check the network again.
+            await response.drain<List<int>>(<int>[]);
+            Isolate.exit(data.port, <Object>[
+              image_provider.NetworkImageLoadException(statusCode: response.statusCode, uri: resolved),
+              data.client,
+            ]);
+          }
+
+          final Uint8List bytes = await consolidateHttpClientResponseBytes(
+            response,
+            onBytesReceived: (int cumulative, int? total) {
+              data.port.send(ImageChunkEvent(
+                cumulativeBytesLoaded: cumulative,
+                expectedTotalBytes: total,
+              ));
+            },
+          );
+          if (bytes.lengthInBytes == 0) {
+            Isolate.exit(data.port, <Object>[Exception('NetworkImage is empty: $resolved'), data.client]);
```

Comment:
@aam my understanding is that since isolate groups, passing data though Isolate.exit should not occur any copying overhead. Is this correct, or should I be using transferrable typedata?

Reply:
Yes, that's correct. Sorry for late response.